packages/js-dapi-client/lib/methods/core/subscribeToTransactionsWithProofsFactory.js

```diff
@@ -39,7 +39,7 @@ function subscribeToTransactionsWithProofsFactory(grpcTransport) {
       ...options,
     };
 
-    if (options.fromBlockHeight === 0) {
+    if ('fromBlockHeight' in options && options.fromBlockHeight === 0) {
       throw new DAPIClientError('Invalid argument: minimum value for `fromBlockHeight` is 1');
     }
```

Comment:
If we added something like this, it could prevent confusion and unexpected behavior. I guess that would be a break change and should target v2.0 instead of master.

```suggestion
    if ('fromBlockHash' in options && 'fromBlockHeight' in options) {
      throw new DAPIClientError('Only one of fromBlockHash or fromBlockHeight may be provided');
    }

    if ('fromBlockHeight' in options && options.fromBlockHeight === 0) {
      throw new DAPIClientError('Invalid argument: minimum value for `fromBlockHeight` is 1');
    }
```

Reply:
0 is default value for integer in protobuf event if this field is omitted.

Reply:
I found/fixed an issue in the code I was trying to use dapi-client with. However, dapi-client's handling of these options is still misleading. Someone should only provide either a height or a hash, but it's possible to provide both and it's obvious that the height is ignored in that case. Wouldn't it make sense to notify the dev they should only be passing one of the two?